z3rro/src/prover.rs

```diff
@@ -21,6 +25,68 @@ pub enum ProveResult<'ctx> {
     Unknown(ReasonUnknown),
 }
 
+/// Find the swine-z3 file located under the dir directory, and execute swine-z3 on the file located at file_path
+fn execute_swine(dir: &Path, file_path: &Path) {
+    let swine = "swine-z3";
+
+    let find_output = Command::new("find")
+        .arg(dir)
+        .arg("-name")
+        .arg(swine)
+        .output().unwrap();
+
+    if find_output.status.success() {
+        let stdout = String::from_utf8_lossy(&find_output.stdout);
+
+        for line in stdout.lines().rev() {
+            let path = Path::new(line);
+
+            if path.exists() && path.is_file() {
+                let cmd_output = Command::new(path)
+                    .arg(file_path) 
+                    .output().unwrap();
+
+                if cmd_output.status.success() {
+                    println!("{}", String::from_utf8_lossy(&cmd_output.stdout));
```

Comment:
The next step would be to parse the output of SWINE and return a [SatResult](https://docs.rs/z3/latest/z3/enum.SatResult.html).